evalml/automl/engine/ray_engine.py

```diff
@@ -0,0 +1,60 @@
+from evalml.automl.engine.engine_base import (
```

Comment:
Ray basically works out-of-the-box! I'm including this mainly as a selling point for the proposed design. I realize ray integration is out-of-scope and could be done later.

![image](https://user-images.githubusercontent.com/41651716/111335943-2cf73600-864b-11eb-874e-3fca02acee28.png)

Reply:
I'll get rid of it before merge!